enclave-protocol/src/lib.rs

```diff
@@ -68,24 +68,24 @@ pub fn is_basic_valid_tx_request(
     request: &VerifyTxRequest,
     tx_inputs: &Option<Vec<SealedLog>>,
     chain_hex_id: u8,
-) -> Result<(), ()> {
+) -> Result<(), std::string::String> {
     if request.info.chain_hex_id != chain_hex_id {
-        return Err(());
+        return Err("hex id mismatch".into());
     }
     match request.tx {
         TxEnclaveAux::DepositStakeTx { .. } => match tx_inputs {
             Some(ref i) if !i.is_empty() => Ok(()),
-            _ => Err(()),
+            _ => Err("sealed log is empty".into()),
         },
         TxEnclaveAux::TransferTx { .. } => match tx_inputs {
             Some(ref i) if !i.is_empty() => Ok(()),
-            _ => Err(()),
+            _ => Err("sealed log is empty".into()),
         },
         TxEnclaveAux::WithdrawUnbondedStakeTx { .. } => {
             if request.account.is_some() {
                 Ok(())
             } else {
-                Err(())
+                Err("request account is empty".into())
```

Comment:
better to return `enum` listing all possible errors rather than a string

Reply:
I'll add a error.rs file and use enum errors